### PR TITLE
Removes Traitor final objectives

### PR DESCRIPTION
**Pull request by Jacquerel**
## About The Pull Request

Astute readers of #79788 suggested that everything I said applied equally to Traitor final objectives.
That is correct.

This PR removes final objectives from Traitors. 
I think Traitors are overall in a better place than Heretics and don't need quite such a substantial redesign, but I don't like this element of their kit.

We have already removed these from midround traitors, but I don't think they're even particularly healthy on roundstart ones (and one round can still start with three or four of those, on top of similarly round-ending heretics).

This PR is arguably even less likely to be merged than the other one because @Watermelon914 has his own ideas about what to do with Final Objectives but I thought I would open it anyway to see what the reception is. If Watermelon still feels sufficiently strongly that they're a good idea, I delegate that to him as his expert area but I'd like to have the discussion.

In terms of the code this actually had less footprint than final objectives (the code is pretty nicely designed), although partially that's because they're a quasi-secret feature which is never explained to you in game.

## Why It's Good For The Game

To summarise what I wrote already in a different PR, the pros of Final Objectives are that they're very cool and the cons are that they have a chilling behavioural effect on players, both traitors and also the crew who react to them.

While it is a shame to remove things that are cool, I think that the negative effect is stronger than the positive, and I am not certain that there is a redesign which would both preserve final objectives _and_ fix the issues I think the concept has.
**Ultimately I think that for a common antagonist, having a pre-written path you progress down until unlocking a round-ending ability actually isn't desirable.**

The existence of the distant reward at the end of a path of tasks creates a powerful incentive for players to pursue it instead of other goals, and ultimately I do not think the gameplay encouraged by this is beneficial.
Additionally, the knowledge among players that treating a traitor with anything other than the maximum possible response is likely to eventually result in a dragon suddenly bursting out of maintenance or the arrival of a battlecruiser promotes an unhealthy relationship between the crew and antagonists who both start performing harsher escalation in response to the other.
This combined with our Dynamic gamemode where there are often _several_ threats which are potentially round-ending (but also might not be, but they're impossible to tell apart) encourages the game's worst aspects of playing to win from all sides.

Several of the more interesting effects (Resonance Cascade and Dark Matter) are actually already achievable in the sandbox without needing a special objective-provided tool to trigger, and I am sure someone can find a way to make Romerol return in a form which doesn't have it show up every round. The others, I will be honest, I will not personally miss all that much.

Personally, when it comes to Traitors, Heretics, and Changelings, I _do_ want them to be capable of ending the round _but I want them to do it themselves_, not simply to be a consequence of just ticking enough boxes.

## Changelog

:cl:
balance: Traitors can no longer receive a final concluding objective, and will merely have to use their 36 telecrystals to oppose the crew instead.
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
How do you get Romerol now as non nuke? Zombie round were one of the funniest in my experience.

**Comment by SimplySmiley123** [comment]
What wil happen to Dark Matter Singularity as well as Resonance Cascade? Being admin-only it will never ever happen again pretty much.

**Comment by SimplySmiley123** [comment]
> Personally, when it comes to Traitors, Heretics, and Changelings, I do want them to be capable of ending the round but I want them to do it themselves, not simply to be a consequence of just ticking enough boxes.

And again, in what way will that ever happen?
Mass murderbone or grand sabotage. 
At least with finals or ascensions you get _some_ sort of warning as crew that shit went south really badly and get a cool moment out of it. Without it it will jsut be back to wordless murderbone and bombing medbay being the maintstay.

**Comment by plutori** [comment]
> What wil happen to Dark Matter Singularity as well as Resonance Cascade? Being admin-only it will never ever happen again pretty much.

Technically the resonance cascade can be done via atmospherics. If you have a PHD in atmospherics and you can create enough hypernob and antinob to cause a singularity delamination in the timespan of one shift. So, you know, all of 3 elder god atmosians can pull it off and it will now be inaccessible for 99.999999% of the playerbase, who will then go on to complain that atmospherics is overpowered and has too high of a learning curve for it's power level and it will either get dumbed down or cascade will get removed, just like what happened with chemistry and chloral hydrate!

**Comment by Jacquerel** [comment]
> What wil happen to Dark Matter Singularity as well as Resonance Cascade? Being admin-only it will never ever happen again pretty much.

Both of those things are actually independently achievable without having the final objective _already_ and resonance cascade doesn't even require you to be a traitor to do it.

**Comment by Fikou** [comment]
> So now we get the time gates of progression traitor but without the final objectives at the end which were the compensation for the time gates? What a terrible PR.

what compensation? they were just things for you to do to end the round. the time gates are for the round health so you dont bomb medbay roundstart, they don't need to give traitors anything in return

**Comment by plutori** [comment]
> You really don't, what you get is that suddenly someone has turned into a dragon with no buildup and then they get lasered to death. In many respects final objectives are actually even more underwhelming in terms of buildup than ascensions, there isn't any particular tell one is going to happen and there especially isn't a tell that any _particular_ one is going to happen.

Except this is wrong. The dark matter singularity gives the crew a global announcement that someone is trying to summon it, and they can run out to space quickly to stop it, and the resonance cascade has global mood debuffs and debuff messages like "Something feels off" or "something feels wrong".

**Comment by Jacquerel** [comment]
> > You really don't, what you get is that suddenly someone has turned into a dragon with no buildup and then they get lasered to death. In many respects final objectives are actually even more underwhelming in terms of buildup than ascensions, there isn't any particular tell one is going to happen and there especially isn't a tell that any _particular_ one is going to happen.
> 
> Except this is wrong. The dark matter singularity gives the crew a global announcement that someone is trying to summon it, and they can run out to space quickly to stop it, and the resonance cascade has global mood debuffs and debuff messages like "Something feels off" or "something feels wrong".

That's actually still true, you have never needed the final objective to summon the dark matter and it performs exactly the same warnings if you don't have it. A normal emag does the job.

**Comment by plutori** [comment]
> > > You really don't, what you get is that suddenly someone has turned into a dragon with no buildup and then they get lasered to death. In many respects final objectives are actually even more underwhelming in terms of buildup than ascensions, there isn't any particular tell one is going to happen and there especially isn't a tell that any _particular_ one is going to happen.
> > 
> > 
> > Except this is wrong. The dark matter singularity gives the crew a global announcement that someone is trying to summon it, and they can run out to space quickly to stop it, and the resonance cascade has global mood debuffs and debuff messages like "Something feels off" or "something feels wrong".
> 
> That's actually still true, you have never needed the final objective to summon the dark matter and it performs exactly the same warnings if you don't have it. A normal emag does the job. The same with a resonance cascade if you trigger one normally. If you want to do either of these to cap off a round you still can.
> 
> This is actually my preference. I **_much_** prefer this kind of thing being part of the sandbox available at any time rather than because you've been assigned a specific objective after performing a bunch of unrelated tasks.

Why do you prefer that an antagonist should spend the entire shift huddled away in atmospherics manufacturing hyper/antinob until they just end the shift by walling off and fortifying engineering and doing a resonance cascade after not interacting with anyone for an hour and a half, rather than having them out running around kidnapping people, assaulting people, and stealing people's eyeballs, which gives security cause to go after them?

Are you SURE that's better for the game?

**Comment by SomeoneYouProbablyKnow** [comment]
> Are you SURE that's better for the game?

Correct. Atmos is bad for the game, but we can't remove it sadly.

**Review by Ghommie — COMMENTED** [review]
Players treat traitors (and several other antagonists), as kill-on-sight in many cases. The amount of disaster that they can bring by simply interacting with some features of the game, on top of their own arsenal and abilities, is itself a reason to such approach, while final objectives oft remain as a sidetrack at the mercy of a late shuttle call.

If final objetives lacking foreshadowing were an issue, so would be what a player can "cook" on their own from what I can tell, but that's, again, the motif of why you aren't safe around traitors. A traitor caught in the open doing secondary objectives will still meet the bad end of the stick final objectives or not. There's little incentive to do otherwise spare the occasional "big brain" play; they're all in all a liability to keep alive.

I can see how they're incompatible on Manuel's more roleplay-friendly scenario, but I haven't heard much of a noise against nor in favor of them on other servers. Most of the time, it all goes downhill before you get a shot at them. Honestly, I find much of what you say sound, especially about the Dynamic gamemode, I just wish this were a confg to spare Manuel from this ill-fitting feature, and leave it to the rest.

File: tgui/packages/tgui/interfaces/Uplink/PrimaryObjectiveMenu.tsx
Comment (Ghommie) [inline tgui/packages/tgui/interfaces/Uplink/PrimaryObjectiveMenu.tsx:1]: Remove line 53

**Review by Ghommie — CHANGES_REQUESTED** [review]
Also, linters again. EDIT: Putting a DNM tag for the time being because it's pretty much on par with the heretic ascension PR.

**Comment by Jacquerel** [comment]
> Also, linters again. EDIT: Putting a DNM tag for the time being because it's pretty much on par with the heretic ascension PR.

I mean Fikou had a specific request for the heretic PR and not for this one 😅 
I believe Watermelon has _also_ provided some input on Discord here but it's not quite the same situation as the other PR.

**Comment by Higgin** [comment]
> There's little incentive to do otherwise spare the occasional "big brain" play; they're all in all a liability to keep alive.

This is hard to encourage as long as there's effectively no ceiling on what antags can do and no enforced commitment to staying below a certain level of violence or disruption. A heretic who says 'I'm just playing with rifts bro' can turn around and kill you fifteen minutes later - good faith is unfortunately a bad play.

Beside that, this and its sibling are good PRs, and this is a good step towards addressing a big problem: progression antags scale in a positive feedback loop. They win so they then win harder. This makes the final objectives or ascension, by the time they appear, much less a matter of drama or climactic tension as a foregone conclusion.

Even if there's no ceiling on what people can do, progression antags' ability to go dummy at basically any threat level in combination with - justified by - this scaling, which has them out kicking down doors, kicking puppies, and accumulating power left and right, massively raises the stakes on dealing with them whether it's a blue star or a black orbit. There's no inherent limit to the threat level that tells players 'play differently' - so they don't, on either side, and it reinforces the most brutal tendencies of play for everyone to the detriment of doing anything less.

I'd suggest that this PR be part of a larger program doing the following: 

1. Put a threat-level limit below which progressive traitors cannot do final objectives (maybe vary it up per final objective? idk, mess around with it.)
2. When somebody rolls progtot, give them a popup that does the following: give them the threat level, tell them what their limits are, and give them a choice of whether or not to accept the roll, refund it, or maybe even trade into some other antag type. 
3. When final objectives are on the table, make them cost TC to buy. Heretic sort of has this with the knowledge cost of your final ritual, but this is sort of a trap if you're not aware of it or pointlessly trivial if you're not pushed up hard against the round timer. You've probably already kicked over enough of the station for it to not matter, but having to actually spend your TC on your final objective makes the choice to forgo other ways of ending the round - your 'mountain to climb' - a meaningful tradeoff against also having an elite MOD + desword + noslips.

**Comment by Xackii** [comment]
You can also delete tritor reputation because it won’t make any sense without a final goal.

**Comment by SomeoneYouProbablyKnow** [comment]
> You can also delete tritor reputation because it won’t make any sense without a final goal.

The main point of reutation and rep locked items is to prevent you from bombing medbay 3 minutes in.

**Comment by Xackii** [comment]
> > You can also delete tritor reputation because it won’t make any sense without a final goal.
> 
> The main point of reutation and rep locked items is to prevent you from bombing medbay 3 minutes in.
> 
> So something like that needs to exist. And rep is good enough for now probs


The only difference between exploding med bay right away or an hour later is that you spend an hour waiting for reputation. 
I think that in an hour it will be easier for engineers to cry on the shuttle than to fix anything

**Comment by Admiral-Gigaton** [comment]
> > What wil happen to Dark Matter Singularity as well as Resonance Cascade? Being admin-only it will never ever happen again pretty much.
> 
> Both of those things are actually independently achievable without having the final objective _already_ and resonance cascade doesn't even require you to be a traitor to do it.
> 
> > And again, in what way will that ever happen? Mass murderbone or grand sabotage. At least with finals or ascensions you get _some_ sort of warning as crew that shit went south really badly and get a cool moment out of it. Without it it will jsut be back to wordless murderbone and bombing medbay being the maintstay.
> 
> You really don't, what you get is that suddenly someone has turned into a dragon with no buildup and then they get lasered to death. In many respects final objectives are actually even more underwhelming in terms of buildup than ascensions, there isn't any particular tell one is going to happen and there especially isn't a tell that any _particular_ one is going to happen.
> 
> > How do you get Romerol now as non nuke? Zombie rounds were one of the funniest in my experience.
> 
> Currently you don't but it would be reasonable to PR adding it back in some fashion, as long as it doesn't become a "purchased every round" sort of deal.

How would I get a dark Matteor, emag all the meteor sats?

**Comment by Jacquerel** [comment]
> The only difference between exploding med bay right away or an hour later is that you spend an hour waiting for reputation. I think that in an hour it will be easier for engineers to cry on the shuttle than to fix anything

there are other people playing the game who have an additional hour before the engineers give up and cry on the shuttle so that sounds like working as intended, the traitor isnt the only person we're building the game for
i dont know why youd wait an hour though when largely the only stuff which takes more than 30 minutes to unlock are badass cosmetics

**Comment by Ghommie** [comment]
> I mean Fikou had a specific request for the heretic PR and not for this one 😅 I believe Watermelon has _also_ provided some input on Discord here but it's not quite the same situation as the other PR.

Okay dokay, guess I'll provide some actual input: Merging this PR as is will leave a bunch of inaccessible code (hardly accessible without var-editing) for things like the battlecruiser cluttering the reposity. Make sure the stuff not used elsewhere is also removed properly unless there's good reason to keep it. I know made the PR overnight.

**Comment by itseasytosee** [comment]
long time coming

**Comment by Xackii** [comment]
> > The only difference between exploding med bay right away or an hour later is that you spend an hour waiting for reputation. I think that in an hour it will be easier for engineers to cry on the shuttle than to fix anything
> 
> there are other people playing the game who have an additional hour to play the game while youre hiding in a locker before the engineers give up and cry on the shuttle so that sounds like working as intended, the traitor isnt the only person we're building the game for i dont know why youd wait an hour though when largely the only stuff which takes more than 30 minutes to unlock are badass cosmetics

Reputation gives you two choices that are not interesting in terms of self-realization: sit in a locker for an hour waiting for points, or perform the same type of tasks with the risk that your game will end with two stun baton attacks. Both of these transform traitor from a creative antagonist into a target-execution machine. I don’t think that a player, knowing that there are antagonists on the server he is visiting, will sit and wait an hour while they do something interesting.

**Comment by Jacquerel** [comment]
> > > The only difference between exploding med bay right away or an hour later is that you spend an hour waiting for reputation. I think that in an hour it will be easier for engineers to cry on the shuttle than to fix anything
> > 
> > 
> > there are other people playing the game who have an additional hour to play the game while youre hiding in a locker before the engineers give up and cry on the shuttle so that sounds like working as intended, the traitor isnt the only person we're building the game for i dont know why youd wait an hour though when largely the only stuff which takes more than 30 minutes to unlock are badass cosmetics
> 
> Reputation gives you two choices that are not interesting in terms of self-realization: sit in a locker for an hour waiting for points, or perform the same type of tasks with the risk that your game will end with two stun baton attacks. Both of these transform traitor from a creative antagonist into a target-execution machine. I don’t think that a player, knowing that there are antagonists on the server he is visiting, will sit and wait an hour while they do something interesting.

Why are you sitting and waiting for an hour?

**Comment by GuillaumePrata** [comment]
> Reputation gives you two choices that are not interesting in terms of self-realization: sit in a locker for an hour waiting for points, or perform the same type of tasks with the risk that your game will end with two stun baton attacks.

There is a third option: Do anything else.
Your job, talk with people, teach new players, gear up for the third minute boning you want to do.
Rep exists to let crew enjoy their round while killing le old "Speedrun bomb medbay any%" we had in the past.

**Comment by Xackii** [comment]
> > Reputation gives you two choices that are not interesting in terms of self-realization: sit in a locker for an hour waiting for points, or perform the same type of tasks with the risk that your game will end with two stun baton attacks.
> 
> There is a third option: Do anything else. Your job, talk with people, teach new players, gear up for the third minute boning you want to do. Rep exists to let crew enjoy their round while killing le old "Speedrun bomb medbay any%" we had in the past.

Delaying the explosion of medbay is not the best way out of this situation. If the idea of ​​adding reputation was only because someone blew up the medbay every round, then something needs to be done with the bombs. No one is stopping the same genius from make  TTV/single tank for 5 minuts of roundstrt and doing the same thing.

**Comment by Higgin** [comment]
@GuillaumePrata 
> There is a third option: Do anything else. Your job, talk with people, teach new players, gear up for the third minute boning you want to do. Rep exists to let crew enjoy their round while killing le old "Speedrun bomb medbay any%" we had in the past.

some folks never saw the days of tot captains regularly taking their own jetpack, walking to engi to release the singulo, and ending the round in under a neat twenty minutes and it shows - the limits on cheap and easy ways to snowball, roundend, and murderbone as fast as you could before are part of allowing more counterplay and balance. it also helps stretch the round out to where people can use more of what the game has on offer by the time a threat has natural access to their powerspikes.

nothing prevents you from making a stunprod, outskilling a secoff, and going on a run at 00:01 besides your own skill issues.

**Comment by TheVekter** [comment]
I think you have more ground to stand on with removing these than you did with ascension (see #79788) but I'm not convinced it'll do anything but make traitors feel less impactful later in the round. I've always liked some of these final objectives - romerol rounds are always fun and tense and rarely ever feel stale, resonance cascades are cool and literally _never happen_ if a traitor doesn't get the objective to do it... I just feel like there's gotta be a better way to accomplish what you want that doesn't involve gutting a bunch of really neat content from traitors.

**Comment by Jacquerel** [comment]
> I think that traitors are one of the features of the station and a player who enters a server with a dynamic of antagonists rolling is ready (or at least should be ready) for the fact that he can be killed/exploded/kidnapped.

I agree and none of those things are being removed. What are you even talking about?

**Comment by Higgin** [comment]
> I think you have more ground to stand on with removing these than you did with ascension (see #79788) but I'm not convinced it'll do anything but make traitors feel less impactful later in the round. I've always liked some of these final objectives - romerol rounds are always fun and tense and rarely ever feel stale, resonance cascades are cool and literally _never happen_ if a traitor doesn't get the objective to do it... I just feel like there's gotta be a better way to accomplish what you want that doesn't involve gutting a bunch of really neat content from traitors.

why I'm about making them something that only exists above a certain threat level and/or something you can just pay TC to get - because they are cool and special in a way that just plasmaflooding atmos or doing gold core/golem spam aren't.

**Comment by Loafin34** [comment]
I don't get this PR. dragon is a weird one, for sure, but all the other final objectives result in cool things happening for the sake of both observers and crewmembers. The first time I actually saw a resonance cascade was magical, and I want this to happen for other people as well. Battlecruiser especially should stay, it's midround nukies as a result of someone meddling with the comms console. Just nerf or buff if you feel it's too powerful.  this feels like when they removed Summon Events from wizards except for threat 100, as all these awesome possibilities get neutered by not appearing naturally.

**Comment by dalmationer** [comment]
Good PR. To copy what I said in the discord chatroom:

> An important balancing part of non prog traitor that i think wasnt really consciously considered is the 'escape' objective. In older versions of the game that were less combat and more social deception focused this incentivised keepimg your cover, but it ALWAYS incentivised, you know, doing something to force a shuttle call when your objectives are done. Progtraitor and heretic kind of go against that philosophy because they encourage doing a long series of tasks and being a constant engine of action until your final objective or ascension. Ive seen both recall the shuttle for more time to do objectives.
> 
> In part i think thats due to the removal of stun combat in favour of better feeling 'fair' combat-- the challenge of older ss13 was that while it was easy to surprise people and kill or kidnap them with stuns, sec could get you with one taser, which really incentivised stealth. Modern ss13 is much more swingy and any fight is likely to have more back and forth... But also antags arent as afraid to go loud.

Personally, I enjoy some of that swinginess, but I feel that final objectives and progtot in general have a tendency to make every traitor a potentially round ending threat and detract from RP. It incentivises a gameplay style of constant action, rather than stopping and coming up with a cool plan which is what I think that Traitor is best for.
It's my personal opinion that traitor should be the gold standard by which other antagonists are balanced, as IMO they, and changeling, are the most freeform and well-integrated ones.

Part of the issue with final objectives, in my opinion, is that they represent a fix for an issue that was created **by** progtot: When you have a set few objectives, you have a set finish point, and you're expected to get the shuttle called and end the round, but can spend some time doing those objectives well and coming up with a plan.
Progtot incentivises always flying by the seat of your pants: you dont know what your next objectives will be and have to improvise, leading to a few, very dominant loadouts (dsword, revolver, etc.) which are best for generalist play. This doesn't have a natural endpoint, though, like having a few objectives and an escape/escape alone/die gloriously objective does. It's something that, as a bit of an oldhead, has bothered me since I started playing the game again.

**Comment by optimumtact** [comment]
> I just wish this were a confg to spare Manuel from this ill-fitting feature, and leave it to the rest.

we dont design for specific servers

**Comment by SimplySmiley123** [comment]
> > > > The only difference between exploding med bay right away or an hour later is that you spend an hour waiting for reputation. I think that in an hour it will be easier for engineers to cry on the shuttle than to fix anything
> > > 
> > > 
> > > there are other people playing the game who have an additional hour to play the game while youre hiding in a locker before the engineers give up and cry on the shuttle so that sounds like working as intended, the traitor isnt the only person we're building the game for i dont know why youd wait an hour though when largely the only stuff which takes more than 30 minutes to unlock are badass cosmetics
> > 
> > 
> > Reputation gives you two choices that are not interesting in terms of self-realization: sit in a locker for an hour waiting for points, or perform the same type of tasks with the risk that your game will end with two stun baton attacks. Both of these transform traitor from a creative antagonist into a target-execution machine. I don’t think that a player, knowing that there are antagonists on the server he is visiting, will sit and wait an hour while they do something interesting.
> 
> Why are you sitting and waiting for an hour? Do you think most non-antagonist players are simply waiting for traitors to do something and are otherwise unstimulated?

Judging by the amount of times crew goes out of their way to hunt down antags the moment they notice them, even if they aren't any particular threat to them, yeah it does feel like crew is understimulated without something happening.
It's even worse if they find out there's nothing going to happen. There are reasons why greenshifts somehow end up more chaotic then your regular orange star, simply put, crew gets bored without _some_ threat to keep them occupied.

And also, honestly, the side objectives are just boring and in wack order when it comes to risk and reward.
Kidnapping people should scale with their role and where the drop-off is. barely anyone if at all bothers iwth the kidnaps obejctives because it's simply not worth bothering to try and for example capture a captain alive to send them off in the SM room at the start of a round (this is the kind of kidnap objective I seen way too many times, and honestly ain't rewarding enough for all the hassle and risk involved with this one for how early on these appear).

**Comment by SimplySmiley123** [comment]
> An important balancing part of non prog traitor that i think wasnt really consciously considered is the 'escape' objective. In older versions of the game that were less combat and more social deception focused this incentivised keepimg your cover, but it ALWAYS incentivised, you know, doing something to force a shuttle call when your objectives are done. Progtraitor and heretic kind of go against that philosophy because they encourage doing a long series of tasks and being a constant engine of action until your final objective or ascension. Ive seen both recall the shuttle for more time to do objectives.

I don't think you played much on LRP, even on MRP, the more stealthy approach is something left to the player to decide on, with some gear that lends into that playstyle. Most players, especially TG players, will ignore stealth and go loud as soon as they feel ready to solo security. Even before progtot that was basically how it went with the "greentext" of doing your objectives and escaping not really mattering to both murderboners and gimmick enjoyers.

**Comment by excalisword** [comment]
I still think every single antagonist with progression should have an endpoint.

**Comment by Deultus** [comment]
if you want to remove the ultimate goal from the traitor, then I suggest you think about the fact that the traitor has rather poorly developed goals, after more than a year his goals and their appearance look like a beta version, lol, there is only one high goal, and in general the goals are  boring and monotonous. it would be fun to see more targets with a high reputation, and it would be nice to see expensive and extremely powerful equipment for a high level of reputation, otherwise there is no point in the reputation system at all

**Comment by SethLafuente** [comment]
This is bad for the same reason as heretic ascensions removal but less integral to traitor over thing but its just removing fun stuff

**Comment by CliffracerX** [comment]
To continue my train of thought from #79788 - this is *also* partially a bandage on a player problem, but also frankly a bandage on the fact that *tot objectives aren't great* in their current state.  The progression from petty theft, senseless murder, and...petnapping, to trying to cause a *Supermatter Cascade* feels...*iffy* to say the least - to speak nothing of the fact the objective pool is incredibly limited and tends to lead to poor play in the name of greentexting, especially on lower population servers.  The most fun antags tend to also be ones who are explicitly *not trying to greentext* and instead following their own gimmick and seeking their own goals; a rework of the current system & expansion of possible objectives could probably go a long way in encouraging people to shake up their tot gameplay and actually try to cause some interesting chaos instead of being the latest incarnation of John Sleepypen with dreams of summoning nukies with Runtime as payment.

Of note, I think Tot improvements also need to come from the *policy side* as much as from giving them more content and goals - giving them leeway and hefty encouragement to just *cause novel new chaos* instead of laser-focusing the greentext finale. 
 Otherwise no amount of code reworks & new content are going to avoid their behaviour eventually ending up stale; just look at any server with Drifting Contractors & the number of "free vacation" gimmicks that come up.  Even though they have a fairly open-ended toolkit, the singular objective tends to lead to uncreative greentext-chasing that just makes the crew grouchy and ready to murderbone from just outside of batong range.

**Comment by K4rlox** [comment]
removing all the fun in the game

**Comment by Admiral-Gigaton** [comment]
> To continue my train of thought from #79788 - this is _also_ partially a bandage on a player problem, but also frankly a bandage on the fact that _tot objectives aren't great_ in their current state. The progression from petty theft, senseless murder, and...petnapping, to trying to cause a _Supermatter Cascade_ feels..._iffy_ to say the least - to speak nothing of the fact the objective pool is incredibly limited and tends to lead to poor play in the name of greentexting, especially on lower population servers. The most fun antags tend to also be ones who are explicitly _not trying to greentext_ and instead following their own gimmick and seeking their own goals; a rework of the current system & expansion of possible objectives could probably go a long way in encouraging people to shake up their tot gameplay and actually try to cause some interesting chaos instead of being the latest incarnation of John Sleepypen with dreams of summoning nukies with Runtime as payment.
> 
> Of note, I think Tot improvements also need to come from the _policy side_ as much as from giving them more content and goals - giving them leeway and hefty encouragement to just _cause novel new chaos_ instead of laser-focusing the greentext finale. Otherwise no amount of code reworks & new content are going to avoid their behaviour eventually ending up stale; just look at any server with Drifting Contractors & the number of "free vacation" gimmicks that come up. Even though they have a fairly open-ended toolkit, the singular objective tends to lead to uncreative greentext-chasing that just makes the crew grouchy and ready to murderbone from just outside of batong range.

Well there is the part of triangulation the weak point and doing a large bomb sized c4 explosion to take out medbay, and thr calling in a threat, bugging high value items that you probably just take with you, infecting people with a deadly disease, spreading propaganda, and THEN the cascade. So it goes from petnapping and theft, to bugging offices and spreading mass propaganda while stealing highly dangerous items, to murdering and placing calling cards while also sabotaging lathes, to being an outright terrorist by hacking comms to call a threat, blowing up comms, and turning a whole department into a crater, and finally, in order of worst to best, turning into a dragon, spreading a zombie disease, being Gordon freeman and causing a resonance cascade, calling in a singularity that eats all living beings, to calling in nukies that tend to be better than actual nukies (by then crew has so much pwr gamer gear because of tot blowing away the crew and station, so it's slightly more balanced in that term) so I feel it's a pretty good about how it progresses. Oh and I forgot to mention the brainwashing of crew and kidnapping of crew. And mass murder of crew if they get a shit final objective, or one that makes it so they CAN or DO (zombies, murdering at cc the people who escape cascade, dragon, nukies

**Comment by Jacquerel** [comment]
As with the other PR I am shelving this idea for now.
@Watermelon914 has some plans for Traitors which... I won't say I think will quite solve my problems in totality but I think they're independently interesting and we should see what they do.

It's interesting to me that this PR was _less_ controversial than the Heretic one (although I can't say it's not controversial 😅) but I guess that makes sense when Final Objectives are a late and fairly non-essential addition while Ascension has been intended as a Heretic's ultimate goal from their addition.

I do suspect we might be removing a couple of _specific_ Final Objectives that I (but also, other members of the team) don't think are very good... and as stupid as it might sound given this PR I do actually have an idea for a new one, if we're keeping the system for the moment 😛

**Comment by PotatoOfChaos** [comment]
Shame it got closed. I'll eagerly be awaiting the revival of this.

**Comment by SimplySmiley123** [comment]
> As with the other PR I am shelving this idea for now. @Watermelon914 has some plans for Traitors which... I won't say I think will quite solve my problems in totality but I think they're independently interesting and we should see what they do. They _should_ at least specifically address the problem of "Final Objectives aren't part of a story and just sort of suddenly happen at the end of a trail of unrelated events".
> 
> It's interesting to me that this PR was _less_ controversial than the Heretic one (although I can't say it's not controversial 😅) but I guess that makes sense when Final Objectives are a late and fairly non-essential addition while Ascension has been intended as a Heretic's ultimate goal from their addition.
> 
> I do suspect we might be removing a couple of _specific_ Final Objectives that I (but also, other members of the team) don't think are very good... and as stupid as it might sound given this PR I do actually have an idea for a new one, if we're keeping the system for the moment 😛

A random, completely random thought that crossed my mind, that prolly would take a good chunk of time to do, would be faction themed objectives, with finals tied to them.
Would be neat to see a Gorlex Marauders agent being the one to call in the battlecruiser and peforming the more violent acts for example.

**Comment by Jacquerel** [comment]
I am afraid that making uplink factions matter for anything more than flavour is an idea I strongly disagree with, I understand why people would want it but I think the effects would largely be negative.

**Comment by SomeoneYouProbablyKnow** [comment]
As long as uplink factions are assigned randomly you shouldn't make em impact gameplay.
If you get to choose them then any gameplay effect is bound to be mininal. Just preventing some item combos (simply find a fellow traitor and trade)

IMO factions should be determined by and have an effect on objectives. Since that's the core of the system. Pick sneaky tasks, get more sneaky tasks and a sneaky final task etc etc. Then you can specialize in what kind of items you get, knowing they'll work for your objectives. But as all things, it's a lof of work.

**Pull request closed without merging**